src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -285,5 +284,9 @@ public override ImmutableArray<CSharpAttributeData> GetAttributes()
         public bool HasEnumeratorCancellationAttribute => _baseParameterForAttributes?.HasEnumeratorCancellationAttribute ?? false;
 
         internal override MarshalPseudoCustomAttributeData? MarshallingInformation => _baseParameterForAttributes?.MarshallingInformation;
+
+        internal override bool IsMetadataOptional => _baseParameterForAttributes?.IsMetadataOptional == true;
```

Comment:
We should verify if this causes a behavior change in any other scenarios, and if so, do we want that behavior change. For example, could this cause parameter default value info to be emitted for some other kind of declaration that we don't want it emitted for.

Reply:
@RikkiGibson Based on @jaredpar comment on the issue, a breaking change is okay here.

One possible code that's going to have a behavioral change:

```csharp
using System;
public static class Program
{
    public static void Main()
    {
        void TestLocalAction(int a = 2)
        {
        }

        MapAction((Action<int>)TestMethodAction);
        MapAction((Action<int>)TestLocalAction);
    }

    public static void TestMethodAction(int a = 2)
    {
    }

    public static void MapAction(Delegate action)
    {
        Console.WriteLine(action.Method.GetParameters()[0].HasDefaultValue);
    }
}
```


I couldn't come up with a more "realistic" scenario.

Reply:
I agree that it is good to "break" this scenario, I just want to make sure that we've enumerated the other places in the compiler where a SynthesizedParameterSymbol can be created with a non-null `_baseParameterForAttributes`, in case this might impact other kinds of symbols.

I'll try and sit down later this afternoon and make sure of it myself. I think local functions might be the only scenario where we set a non-null value for this field, although lambdas might do it too now (cc @cston), and we probably would want the same kind of behavior for lambdas as we want for local functions here.

Reply:
I think the only place for `_baseParameterForAttributes` to be non-null is `SynthesizedClosureMethod` (either a lambda or a local function).

Reply:
> I think the only place for _baseParameterForAttributes to be non-null is SynthesizedClosureMethod (either a lambda or a local function).

It feels like we might want to add negative tests for other scenarios consuming ```SynthesizedParameterSymbol.Create```.

Reply:
@AlekseyTs Any concrete/specific test cases you have in mind? For example, top-level entry point and record copy ctor are using `SynthesizedParameterSymbol.Create`. How a test for them would look like? they're never synthesized with optional parameters.